core/string.cpp

```diff
@@ -152,13 +152,17 @@ template <class Key, class Var>
 auto format_variable(const Key &key, const Var &variable,
                      const Dimensions &datasetDims = Dimensions()) {
   std::stringstream s;
+  const auto dtype = variable.dtype();
   const std::string colSep("  ");
   s << tab << std::left << std::setw(24) << to_string(key);
   s << colSep << std::setw(9) << to_string(variable.dtype());
   s << colSep << std::setw(15) << '[' + variable.unit().name() + ']';
   s << colSep << make_dims_labels(variable, datasetDims);
-  s << colSep
-    << apply<ValuesToString>(variable.data().dtype(),
+  s << colSep;
+  if (dtype == DType::PyObject)
```

Comment:
Proper fix would involve adding `pybind11::object` to the tuple [here](https://github.com/scipp/scipp/blob/master/core/include/scipp/core/tag_util.h#L48-L51) but that would involve bringing the pybind11 dependency into scipp core. Maybe there is a better way to do this, a separate `GenericObject` proxy that lives in `scipp::core`?